### PR TITLE
Add support for Trailers

### PR DESCRIPTION
**Pull request by sam-kleiner**
Original discussion: #1593
Issue: #1596

### CODE REVIEW

**Comment by sam-kleiner** [comment]
Looks like I need to update asgi ref before the tests will pass here. I will PR that to add `trailers` property to [HTTPResponseStartEvent](https://github.com/django/asgiref/blob/main/asgiref/typing.py#L98), then also add a new `HTTPResponseTrailersEvent`

**Comment by Kludex** [comment]
> Looks like I need to update asgi ref before the tests will pass here. I will PR that to add `trailers` property to [HTTPResponseStartEvent](https://github.com/django/asgiref/blob/main/asgiref/typing.py#L98), then also add a new `HTTPResponseTrailersEvent`

See https://github.com/django/asgiref/issues/141#issuecomment-1111289069 for more information on what is needed.

File: docs/index.md [hunk 1]
```diff
@@ -459,6 +459,47 @@ async def app(scope, receive, send):
     })
 ```
 
+### HTTP Trailers
```
Comment (Kludex) [inline docs/index.md:462]: Do you think it makes sense to be its own section on a new file called "extensions.md"?

**Comment by Kludex** [comment]
@sam-kleiner Are you willing to keep working on this?

Also, do you have a real use-case for this extension?

**Comment by Kludex** [comment]
Closing as stale.

If someone has a good use case, please create a discussion pointing that out. :pray:

**Pull request closed without merging**